components/data_field.py

```diff
@@ -81,3 +80,35 @@ def draw_base(self, canvas: Canvas):
     def draw_data(self, canvas: Canvas, data: Data):
         self.title = "GPS KPH" if data["gps_speed"] else "REED KPH"
         super().draw_data(canvas, data)
+
+
+class VoltageField(DataField):
+    """ A specialised version of DataField which displays the voltage adjusting
+        for it's values whether the battery is low, medium or high."""
+
+    def __init__(self, title: str,
+                 value_func: Callable[[Data], str],
+                 coordinate: Tuple[int, int],
+                 is_title_static: bool = True):
+        super().__init__(title, value_func, coordinate, is_title_static)
+
+    def draw_base(self, canvas: Canvas):
+        return super().draw_base(canvas)
+
+    def draw_data(self, canvas: Canvas, data: Data):
+        if not self.is_title_static:
+            DataField.draw_base(self, canvas)
+        voltage_value = self.value_func(data)
+        if voltage_value == '--' or float(voltage_value) >= 7.3:
+            colour = Colour.white
+        elif float(voltage_value) >= 7.0:
```

Comment:
All looks good, but just a bit confused about these values. I think the batteries we have only go up to 4.2V and maybe we want to show it as yellow when it's above 3.3V but below 3.7V because I think the midway point is somewhere in that range. @hallgchris will be able to confirm better

Reply:
We use two 3.7V battery cells, so the voltage we get for the overall package is double.